pyproject.toml

```diff
@@ -0,0 +1,130 @@
+[build-system]
+requires = ["setuptools>=64", "setuptools_scm[toml]>=6.2", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "event-model"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+description = "One line description of your module"
+dependencies = [
+    "importlib-metadata",
+    "jsonschema",
+    "numpy",
+    "packaging"
+]
+dynamic = ["version"]
+license.file = "LICENSE"
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+dev = [
+    "dask[array]",
+    "black",
+    "mypy",
+    "flake8",
+    "flake8-isort",
+    "Flake8-pyproject",
+    "pipdeptree",
+    "pre-commit",
+    "pydata-sphinx-theme>=0.12",
+    "pytest",
+    "pytest-cov",
+    "sphynx",
+    "sphinx-autobuild",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "tox-direct",
+    "types-mock",
+
+    # These are dependencies of various sphinx extensions for documentation.
+    "ipython",
+    "matplotlib",
+    "numpydoc",
+    "sphinx-copybutton",
+]
+
+[project.scripts]
+# event_model = "event_model.__main__:main"
+
+[project.urls]
+GitHub = "https://github.com/bluesky/event-model"
+
+#[[project.authors]] # Further authors may be added by duplicating this section
+# email = "email@email.com"
+# name = "Firstname Lastname"
+
+
+[tool.mypy]
+# Ignore missing stubs in imported modules
+ignore_missing_imports = true
+
+
+[tool.isort]
+float_to_top = true
+profile = "black"
```

Comment:
We'd use the pyproject.toml for the black profile `[tool.black]` (if we weren't using defaults).

Reply:
Yes, this is telling isort to use "black-style" `from x import`s (i.e. all on one line if it fits in 88 chars, or one import per line if it doesn't)

Reply:
I was concerned as I saw a max line length setting of 115 elsewhere in the files. I think `black`'s config should match that - therefore I was surprised not to see any `black`'s config in that file.

That's what we use in other repos and in our cookiecutter: https://github.com/NSLS-II/scientific-python-cookiecutter/blob/af48f308a0a81f5ee7ae88abba5f393408bdd262/%7B%7B%20cookiecutter.repo_name%20%7D%7D/pyproject.toml#L2. Should this part be added too?

Reply:
We specifically left the black line length at its default (88) in other projects. All the files have been formatted with that length here. Is there strong feeling that 115 is better for the whole of bluesky?

Reply:
I know that `bluesky-queueserver` was linted with the line length=115, but I don't have a strong feeling for other repos. Probably it'd make all other repos more consistent, but we should discuss it. I am OK with the current setup for this repo to avoid further delays.